### PR TITLE
pybind: Rework cephfs/setup.py for PyPI

### PR DESCRIPTION
**Pull request by onyb**
Also use `set_include_dirs` in `distutils.ccompiler` instead of `add_include_dir` for supporting non-standard include directories.

### CODE REVIEW

File: src/pybind/cephfs/setup.py [hunk 1]
```diff
@@ -1,56 +1,196 @@
...
+except ImportError:
+    print("WARNING: Cython is not installed.")
+
+    if not os.path.isfile('cephfs.c'):
+        print('ERROR: Cannot find Cythonized file cephfs.c', file=sys.stderr)
+        sys.exit(1)
+    else:
+        def cythonize(x, **kwargs):
+            return x
+
```
Comment (tchaikov) [inline src/pybind/cephfs/setup.py]: please remove at least one blank line here.

and could you run this script thru pep8 to kill some warnings.

**Comment by tchaikov** [comment]
lgtm other than a pep8 nit.

**Comment by onyb** [comment]
@tchaikov Thanks. PEP8ified.

**Comment by jdurgin** [comment]
@onyb looks like the build bot failure is real - it fails to find librados.h with a cmake build (happens on my machine too).

If you haven't built with cmake yet, you should start using it - we're just about all switched over. ceph.git's README.md has updated info on using it.

**Comment by onyb** [comment]
@jdurgin Fixed the issue. I've tested this with both CMake and configure.

**Comment by jcsp** [comment]
This didn't cause any issues in this run: http://pulpito.ceph.com/jspray-2016-07-19_05:56:13-fs-wip-jcsp-testing-20160719-distro-basic-mira/

Everyone happy to merge?

**Comment by jdurgin** [comment]
@jcsp I'm happy to merge

**Comment by dillaman** [comment]
Looks like the doc builders are broken -- the include paths from the environment are no longer processed.

**Pull request merged**